### PR TITLE
chore: update FP as signer

### PR DESCRIPTION
**Pull request by RafilxTenfen**
- Update to use FP as signer
- related https://github.com/babylonchain/babylon/pull/678

### CODE REVIEW

File: itest/babylon_node_handler.go [hunk 1]
```diff
@@ -14,42 +14,27 @@ import (
...
+func newBabylonNode(dataDir, walletName string, cmd *exec.Cmd) *BabylonNode {
+	return &BabylonNode{
+		DataDir:    dataDir,
+		cmd:        cmd,
+		WalletName: walletName,
 	}
 }
 
-func (n *babylonNode) start() error {
+func (n *BabylonNode) start() error {
```
Comment (SebastianElvis) [inline itest/babylon_node_handler.go:32]: Why is this needed? same for other functions

**Review by SebastianElvis — APPROVED** [review]

**Review by gitferry — APPROVED** [review]
LGTM! Just a note that the dependency on Babylon should be changed after https://github.com/babylonchain/babylon/pull/678 is merged

File: tools/go.mod [hunk 2]
```diff
@@ -2,7 +2,7 @@ module github.com/babylonchain/vigilante/tools
 
 go 1.22.3
 
-require github.com/babylonchain/babylon v0.9.0-rc.1
+require github.com/babylonchain/babylon v0.8.6-0.20240628205418-d51dc2ba6a4c
```
Comment (KonradStaniec) [inline tools/go.mod:5]: after merging your pr on Babylon side, you can tag `v0.9.0-rc.2` and use it here. We should slowly move away from using hashes

**Review by KonradStaniec — APPROVED** [review]

File: itest/babylon_node_handler.go (hunk 1 above)
Comment (RafilxTenfen) [reply itest/babylon_node_handler.go:32]: to expose the structure for another packages to use functions like `TxBankMultiSend, TxBankSend` when needed
like https://github.com/babylonchain/btc-staker/blob/76c2d4eb65f3a58d612e251eae17470b8ecd98f2/itest/e2e_test.go#L188

File: tools/go.mod (hunk 2 above)
Comment (RafilxTenfen) [reply tools/go.mod]: done at https://github.com/babylonchain/btc-staker/pull/178/commits/fae2d99e6631514d22f74119d1b45e0ceb41e56b
